crates/cubecl-opt/src/block.rs

```diff
@@ -0,0 +1,53 @@
+use std::{cell::RefCell, collections::HashSet, rc::Rc};
+
+use cubecl_core::ir::{Operation, Variable};
+use petgraph::graph::NodeIndex;
+use stable_vec::StableVec;
+
+use crate::{version::PhiInstruction, ControlFlow, Optimizer, Program};
+
+#[derive(Default, Debug, Clone)]
+pub struct BasicBlock {
+    pub phi_nodes: Rc<RefCell<Vec<PhiInstruction>>>,
+    pub(crate) writes: HashSet<(u16, u8)>,
+    pub(crate) live_vars: HashSet<(u16, u8)>,
+    pub(crate) dom_frontiers: HashSet<NodeIndex>,
+    pub ops: Rc<RefCell<StableVec<Operation>>>,
+    pub control_flow: Rc<RefCell<ControlFlow>>,
+}
+
+impl Optimizer {
+    pub fn visit_all(
+        &mut self,
+        mut visit_read: impl FnMut(&mut Self, &mut Variable) + Clone,
+        mut visit_write: impl FnMut(&mut Self, &mut Variable) + Clone,
+    ) {
+        for node in self.program.node_indices().collect::<Vec<_>>() {
+            let phi = self.program[node].phi_nodes.clone();
+            let ops = self.program[node].ops.clone();
```

Comment:
If it's only one operation, I would call the variable `op` instead of `ops`.

Reply:
It's the set of all operations in the block. So it's multiple. We iterate through it below.